src/iuse_actor.cpp

```diff
@@ -1095,6 +1095,14 @@ static ret_val<tripoint> check_deploy_square( Character *p, item &it, const trip
                                                 _( "You attempt to become one with the %s.  It doesn't work." ), it.tname() );
     }
 
+    if( pnt.z() > 1 ) {
```

Comment:
You get this message if you try to deploy the anything above the ground floor - I suggest look at how iexamine::ledge determines the number of levels that are open below a point, and check that instead.
https://github.com/CleverRaven/Cataclysm-DDA/blob/8c9ca7a48cc6855311fa53f89fabb6aa9ab879c3/src/iexamine.cpp#L5530-L5537

Reply:
Because the solution will tell me that a ladder I deploy in a stable situation on a roof is going to fall several stories when it is not.

Reply:
Thanks for the review, issues resolved in [514a65e](https://github.com/CleverRaven/Cataclysm-DDA/pull/77083/commits/514a65eea5749094c823a6bd97fddc5d1dbaff22).